tst/com/amazon/kinesis/streaming/agent/processing/processors/DataConverterTest.java

```diff
@@ -195,12 +195,15 @@ public void testSysLogDataConverter() throws Exception {
         final String expectedLogWithoutPID = "{\"timestamp\":\"Mar 12 12:01:02\",\"hostname\":\"server4\",\"program\":\"snort\",\"processid\":null,\"message\":\"Ports to decode telnet on: 21 23 25 119\"}\n";
         verifyDataConversion(converter, logWithoutPID.getBytes(), expectedLogWithoutPID.getBytes()); 
     }
-    
-    private void verifyDataConversion(IDataConverter converter, byte[] dataBin, byte[] expectedBin) throws Exception {
-        ByteBuffer data = ByteBuffer.wrap(dataBin);
-        ByteBuffer res = converter.convert(data);
-        byte[] resBin = new byte[res.remaining()];
-        res.get(resBin);
-        Assert.assertEquals(resBin, expectedBin);
```

Comment:
Assert.assertEquals should be able to check the two arrays by a byte-to-byte comparison. Are you sure this fails the test case? Otherwise none of the test cases would have succeed.